dataprofiler/profilers/profile_builder.py

```diff
@@ -51,56 +52,82 @@ def __init__(self, df_series, sample_size=None, min_sample_size=5000,
-                          "All statistics will be based on this subsample and "
-                          "not the whole dataset.".format(sample_size))
-        clean_sampled_df, base_stats = \
-            self.get_base_props_and_clean_null_params(
-                df_series, sample_size, sample_ids=sample_ids)
-        self._update_base_stats(base_stats)
+            clean_sampled_df, base_stats  = \
+                self.clean_data_and_get_base_stats(
+                    df_series=df_series, sample_size=sample_size,
+                    min_true_samples=self._min_true_samples, sample_ids=sample_ids)
+            self.update_column_profilers(clean_sampled_df, pool)
+            self._update_base_stats(base_stats)
+
+    def update_column_profilers(self, clean_sampled_df, pool):
+        """
+        Calculates type statistics and labels dataset
+        
+        :param clean_sampled_df: sampled series with none types dropped
+        :type clean_sampled_df: Pandas.Series
+        :param pool: pool utilized for multiprocessing
+        :type pool: multiprocessing.pool
+        """
 
-        self.profiles = {
-            'data_type_profile':
-            ColumnPrimitiveTypeProfileCompiler(
-                clean_sampled_df, self.options, pool),
-            'data_stats_profile':
-            ColumnStatsProfileCompiler(
-                clean_sampled_df, self.options, pool)
-        }
+        if self.name is None:
+            self.name = clean_sampled_df.name
+        if self.name != clean_sampled_df.name:
+            raise ValueError(
+                'Column names have changed, col {} does not match prior name {}',
+                clean_sampled_df.name, self.name
+            )
+        
+        # First run, create the compilers
+        if self.profiles is None or len(self.profiles) == 0:
+            self.profiles = {
+                'data_type_profile':
+                ColumnPrimitiveTypeProfileCompiler(
+                    clean_sampled_df, self.options, pool),
+                'data_stats_profile':
+                ColumnStatsProfileCompiler(
+                    clean_sampled_df, self.options, pool)
+            }
+        
+            use_data_labeler = True
+            if self.options and isinstance(self.options, StructuredOptions):
+                use_data_labeler = self.options.data_labeler.is_enabled
+
+            if use_data_labeler:
+                self.profiles.update({
+                    'data_label_profile':
+                    ColumnDataLabelerCompiler(
+                        clean_sampled_df, self.options, pool)
+                })
+        else:
 
-        use_data_labeler = True
-        if options and isinstance(options, StructuredOptions):
-            use_data_labeler = options.data_labeler.is_enabled
+            # Profile compilers being updated
+            for profile in self.profiles.values():
+                profile.update_profile(clean_sampled_df, pool)
```

Comment:
I see there were some indents, but I am confused bc the if statement creates profiles if they don't exist, then why not just update them since they now exist?

Reply:
That's what's happening?

Reply:
it is inside an else statement contrary to creating the profiles:
```
if self.profiles is None or len(self.profiles) == 0:
    # create profiles
else:
    # Profile compilers being updated
    for profile in self.profiles.values():
        profile.update_profile(clean_sampled_df, pool)
```